### ISSUE TITLE
Bug: [sort-type-union-intersection-members] Fix for constituents with leading pipes or ampersands produces invalid output

### ISSUE THREAD

**Issue by cherryblossom000** [typescript-eslint/typescript-eslint#6117]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBAKlC8UAUAfKBvAvgSigMg0wCgg&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZAe2jqNNtiaUOTWs3yJoyRGXxCRAW0TyARhNQYATODABfEDqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type T = (| {}) & {}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/sort-type-union-intersection-members": 2
  },
};
```

### Expected Result

`eslint --fix` should result in valid TypeScript output:

```ts
type T = {} & (| {})
```

### Actual Result

`eslint --fix` results in this invalid code:

```ts
type T = {} & | {}
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.44.0` |
| `@typescript-eslint/parser`            | `5.44.0` |
| `TypeScript`                           | `4.9.3` |
| `ESLint`                               | `8.28.0` |
| `node`                                 | `18.12.1` |
